sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -318,11 +315,56 @@ private static void validateTaskRequirements(List<TaskRequirement> taskRequireme
             }
         }
 
-        return resources;
+        return coalescePorts(resources);
+    }
+
+    private static List<Protos.Resource> coalescePorts(Collection<Protos.Resource> resources) {
+        // Within the SDK, each port is handled as its own resource, since they can have extra meta-data attached, but
+        // we can't have multiple "ports" resources on a task info, so we combine them here. Since ports are also added
```

Comment:
It's meant to return a list. The use case is this: we have a `ResourceSpecification` per resource type usually, but in the case of ports there's a specification per port, so a naive translation of spec to proto will give you a list of `Resource` that has multiple port-type resources in it. This just returns a list of resources where all the ports have been coalesced into a single `Resource` (slightly misleading, since it actually just clears the ranges, but that's the idea).